pkg/karmadactl/cmdinit/kubernetes/deploy.go

```diff
@@ -283,15 +284,33 @@ func (i *CommandInitOption) Complete() error {
 	return initializeDirectory(i.KarmadaDataPath)
 }
 
-// initializeDirectory initializes a directory and makes sure it's empty.
+// initializeDirectory ensures that the specified path is a directory.
+// If the path does not exist, it creates the directory with the specified permissions (0700).
+// If the path exists as a file, it returns an error indicating that the path is a file, not a directory.
+// If the path exists as a directory, it removes the existing directory and its contents before creating a new one.
+//
+// Parameters:
+//
+//	path (string): The path to check or create.
+//
+// Returns:
+//
+//	error: An error if the path exists as a file, or if there was an issue creating the directory
+//	       or removing an existing directory, or nil if successful.
 func initializeDirectory(path string) error {
-	if utils.IsExist(path) {
+	pathType, exists := karmadactlutil.IsExist(path)
+	if exists && pathType == karmadactlutil.Directory {
 		if err := os.RemoveAll(path); err != nil {
-			return err
+			return fmt.Errorf("failed to remove existing directory '%s': %w", path, err)
```

Comment:
When a path exists, could we only return the path's information without removing files in it, allowing the caller to decide how to handle it? This would make the function more cohesive. Additionally, the `register` command does not require the deletion of existing files in the directory.

Reply:
Sorry, I noticed that you implemented the `initializeDirectory`  in command `init` and `register` as two separate functions. Could these be combined into a single function?